### ISSUE TITLE
`has_num` and `has_den` encourage bad practice

### ISSUE THREAD

**Issue by johelegp** [nholthaus/units#115]
The problem with `has_num` and `has_den` is that they are hardly useful on their own, yet are part of the public interface of the library. It is like the `Addable` concept which only tests for the presence of a binary `operator+`, when what we really want is a `Number`. `std::string` matches `Addable`, but not with the expected semantics of a `Number`.

We already have what we really want, `is_ratio_v`, so I suggest dropping `has_num` and `has_den` from the public interface for v3.0.0. I'm already working on some patches which include these changes, but I'm opening an issue anyways. The patch is a simplification of `is_ratio` that depends on the library relying only on `std::ratio` and not a `Ratio` implementation anymore which I found digging through the commits.

**Comment by nholthaus** [comment]
@johelegp agree. Appreciate both the issues and the patches. Thanks for pushing me and the library to take the quality to the next level.

Really wish we had an `std::is_ratio_v` though...

**Comment by nholthaus** [comment]
Done: `has_num`/`has_den` were removed from the public interface (use `is_ratio_v`). Closing.
